### PR TITLE
Add interests dropdown to UI and modify components to establish consistency in style and backend JSON values

### PR DESCRIPTION
**Pull request by dhruvi618**
Files changed: 
- AddParticipantServlet.java: change JSON const string to match updated payload value (timeAvailableUntil --> endTimeAvailable)
- DurationDropdown.js: Add style changes to dropdown component 
- App.js: Change heading elements to be paragraph elements for style purpose (`<h4> to <p>` and `<h2> to <h3>`)
- Form.js: Change JSON payload key from timeAvailableUntil to endTimeAvailable and change button text from "Submit" to "Match Me!"
- InterestsDropdown.js: Add multiple select dropdown for user to select interests and corresponding onChange methods to keep track of state
- RoleDropdown.js: Add additional roles at Google and add style changes along with replacing `MenuItem` and `ListItemText` to HTML `option` for consistency among the style of dropdowns
- ProductAreaDropdown.js: Add style changes and replace `MenuItem` and `ListItemText` to HTML `option` for consistency among the style of dropdowns
- MatchPreference.js: Change value from `none` to `any` for consistency with backend and change order of `@param` and `@return` in function JSdoc 
- PreferenceCheckbox.js: Change order of `@param` and `@return` in function JSdoc 
- index.css: Change global font-family to Manrope - sans serif for style purposes

Test files changed:
- InterestsDropdown.test.js: Add snapshot UI tests and unit tests to ensure rendering of interests in the correct alphabetical order for the `<InterestsDropdown />` component

Snapshot files changed: 
- App.test.js.snap: Update snapshot file since the `<App />` component's child components were updated
- Form.test.js.snap: Update snapshot file since the `<Form />` component's child components were updated
- MatchPreference.test.js.snap: Update snapshot file since the `<MatchPreference />` component's value field and JSX were altered
- DurationDropdown.test.js.snap: Update snapshot file since the `<DurationDropdown />` component's style and JSX were altered
- InterestsDropdown.test.js.snap: Create snapshot file for the `<InterestsDropdown />` component as it's corresponding test file was created

Project files changed (dependencies):
- package.json: Add enzyme testing library and enzyme adapter dependencies 
- yarn.lock: Add testing library dependencies and update existing plugins/dependencies

### CODE REVIEW

File: frontend/src/components/InterestsDropdown.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import renderer from 'react-test-renderer';
+
+describe('Interests Dropdown', () => {
+  it('should be defined', () => {
+    expect(InterestsDropdown).toBeDefined();
+  });
+  it('should render correctly', () => {
+    const tree = renderer.create(<InterestsDropdown />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
```
Comment (kmh287) [inline frontend/src/components/InterestsDropdown.test.js:12]: it should be populated with a list of interests

it should be populated in alphabetical order

**Comment by dhruvi618** [comment]
> LGTM. As a general comment, I think it'll improve the efficiency to review this PR if it's broken into two: one to refactor (as I see some renaming in java / js and use select instead of native select, etc.), and the other to add new components like interests dropdown and add more roles.
> 
> Also, it might be interesting to include a screenshot of the new UI with each UI changes.

Deployed the changes to http://ad-lib-step-2020.appspot.com/

**Comment by dhruvi618** [comment]
> LGTM! Love the large font size! WDYT about unbolding the intro blurb at the top? Also what's the status of the font and color change?

Thank you! I will add the unbold change to this PR and I have the font and color change modifications on my to-do list for after I create all the views and connect them so I can establish a global theme!

File: frontend/src/components/InterestsDropdown.test.js (hunk 1 above)
Comment (dhruvi618) [reply frontend/src/components/InterestsDropdown.test.js:12]: Thank you for the suggestion, changed!

Comment (kmh287) [reply frontend/src/components/InterestsDropdown.test.js:12]: It looks like the first suggested test didn't make it in. The alphabetical order test won't prevent somebody from changing the dropdown order to non-alphabetic order as long as the new snapshot is approved. I would expect this test to grab the list of interests from the dropdown and assert that they're already sorted.

Comment (dhruvi618) [reply frontend/src/components/InterestsDropdown.test.js:22]: Added both tests, thanks!

File: frontend/src/components/InterestsDropdown.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  it('should be populated with 20 interests', () => {
+    expect(menuItems).toHaveLength(20);
+  });
```
Comment (kmh287) [inline frontend/src/components/InterestsDropdown.test.js:25]: 20 might be too specific and will break if anything is added or removed. How about asserting it has more than zero?

**Pull request merged**